installer/install.go

```diff
@@ -0,0 +1,397 @@
+	ConnectorNamespace 			= "astra-connector"
+	ConnectorOperatorNamespace  = "astra-connector-operator"
+)
+
+func main() {
+	formatter := &log.TextFormatter{
+		FullTimestamp: true,
+	}
+	log.SetFormatter(formatter)
+
+	var opts Options
+	var connectorConfig ConnectorConfig
+	_, err := flags.Parse(&opts)
+	checkFatalErr(err)
+
+	log.Info("Installing Astra Connector")
+
+	absPath, err := filepath.Abs(ConnectorDefaultsConfigPath)
+	checkFatalErr(err)
+	yamlFile, err := ioutil.ReadFile(absPath)
+	checkFatalErr(err)
+	err = yaml.Unmarshal(yamlFile, &connectorConfig)
+	checkFatalErr(err)
+
+	// input check
+	if opts.ImageTar != "" || opts.ImageRepo != "" {
+		if !(opts.ImageTar != "" && opts.ImageRepo != "") {
+			log.Fatal("error: 'image-tar' and 'image-repo' options must be used together")
+		}
+	}
+
+	// input check
+	if opts.HostAlias {
+		if opts.HostAliasIP == "" {
+			log.Fatal("error: 'hostAliasIP' must be defined if the 'hostAlias' flag is set ")
+		}
+	}
+
+	if opts.ImageTar != "" && opts.ImageRepo != "" {
+		dockerClient, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+		checkFatalErr(err)
+
+		log.Info("Loading image tar")
+		loadedImages, err := loadImageTar(dockerClient, opts)
+		checkFatalErr(err)
+
+		log.Info("Tagging images")
+		taggedImages, err := tagImages(dockerClient, loadedImages, opts.ImageRepo)
+		checkFatalErr(err)
+
+		//todo: Fix the auth issue preventing this from succeeding
+		//log.Info("Pushing images")
+		//err = pushImages(dockerClient, &opts, taggedImages)
+		//checkFatalErr(err)
+
+		// This is a workaround until we can push using the docker package
+		for _, image := range taggedImages {
+			pushCmd := exec.Command("docker", "push", image)
+			output, err := pushCmd.CombinedOutput()
+			if err != nil {
+				msg := fmt.Sprintf(fmt.Sprint(err) + ": " + string(output))
+				log.Fatal(msg)
+			}
+			log.Info(string(output))
+		}
+
+		connectorConfig.Spec.ImageRegistry.Name = opts.ImageRepo
+	}
+
+	// Update yaml data
+	connectorConfig.Metadata.Namespace = ConnectorNamespace
+	connectorConfig.Spec.Astra.AcceptEula = opts.AcceptEula
+	connectorConfig.Spec.Astra.AccountId = opts.AstraAccountId
+	connectorConfig.Spec.Astra.ClusterName = opts.ClusterName
+	connectorConfig.Spec.Astra.Token = opts.Token
+	connectorConfig.Spec.NatssyncClient.SkipTLSValidation = opts.SkipTlsValidation
+	connectorConfig.Spec.NatssyncClient.CloudBridgeUrl = opts.AstraUrl
+	connectorConfig.Spec.NatssyncClient.HostAliasIP = opts.HostAliasIP
+	connectorConfig.Spec.NatssyncClient.HostAlias  = opts.HostAlias
+
+
+	// Create namespaces
+	log.Info("Creating Astra Controller namespace")
+	output, err := createNamespace(ConnectorNamespace)
+	checkFatalErr(err)
+	log.Info(output)
+
+	log.Info("Creating Astra Controller Operator namespace")
+	output, err = createNamespace(ConnectorOperatorNamespace)
+	checkFatalErr(err)
+	log.Info(output)
+
+	// Install Astra Controller Operator
```

Comment:
There are multiple places where it says Astra Controller. Did you mean to say Astra Connector?

Reply:
Oops. Yes. Thanks